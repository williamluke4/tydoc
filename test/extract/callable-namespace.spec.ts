it('a callable interface is marked as a callable namespace', () => {
  expect(
    ctx.extract(`
      export interface A {
        (): boolean
        a: 1
        b: 2
      }
    `)
  ).toMatchSnapshot()
})
