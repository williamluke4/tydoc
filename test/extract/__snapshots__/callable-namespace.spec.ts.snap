// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a callable interface is marked as a callable namespace 1`] = `
Object {
  "modules": Array [
    Object {
      "isMain": true,
      "kind": "module",
      "location": Object {
        "absoluteFilePath": "/src/a.ts",
      },
      "mainExport": null,
      "name": "a",
      "namedExports": Array [
        Object {
          "isTerm": false,
          "isType": true,
          "kind": "export",
          "name": "A",
          "type": Object {
            "kind": "typeIndexRef",
            "link": "(a).A",
          },
        },
      ],
      "path": "/",
      "tsdoc": null,
    },
  ],
  "typeIndex": Object {
    "(a).A": Object {
      "hasProps": true,
      "isOverloaded": false,
      "kind": "callable",
      "props": Array [
        Object {
          "kind": "prop",
          "name": "a",
          "type": Object {
            "base": "number",
            "kind": "literal",
            "name": "1",
          },
        },
        Object {
          "kind": "prop",
          "name": "b",
          "type": Object {
            "base": "number",
            "kind": "literal",
            "name": "2",
          },
        },
      ],
      "raw": Object {
        "nodeFullText": "export interface A {
  (): boolean;
  a: 1;
  b: 2;
}",
        "nodeText": "export interface A {
  (): boolean;
  a: 1;
  b: 2;
}",
        "typeText": "A",
      },
      "sigs": Array [
        Object {
          "kind": "sig",
          "params": Array [],
          "return": Object {
            "kind": "primitive",
            "type": "boolean",
          },
        },
      ],
    },
  },
}
`;
